httpx/models.py

```diff
@@ -85,10 +85,17 @@
     str,
     typing.Union[
         typing.IO[typing.AnyStr],  # file
-        typing.Tuple[str, typing.IO[typing.AnyStr]],  # (filename, file)
         typing.Tuple[
-            str, typing.IO[typing.AnyStr], str
-        ],  # (filename, file, content_type)
+            typing.Optional[str],
+            typing.Union[
+                typing.IO[typing.AnyStr], typing.AnyStr
+            ],  # (filename, file (or str))
+            typing.Tuple[
+                typing.Optional[str],
+                typing.Union[typing.IO[typing.AnyStr], typing.AnyStr],
+                typing.Optional[str],
+            ],  # (filename, file (or str), content_type)
+        ],
```

Comment:
I don't think these annotations are correct, are they? If we want to allow a file, a (optional filename, file/str), or an (optional filename, file/str, content_type), this should probably be:

```python
typing.Union[
    # file
    typing.IO[typing.AnyStr],
    # (filename, file (or str))
    typing.Tuple[
        typing.Optional[str],
        typing.Union[typing.IO[typing.AnyStr], typing.AnyStr],
    ],
    # (filename, file (or str), content_type)
    typing.Tuple[
        typing.Optional[str],
        typing.Union[typing.IO[typing.AnyStr], typing.AnyStr],
        typing.Optional[str],
    ],
]
```

Reply:
When I diffed your annotation with the current ones they seemed to be identical, just different comment positions? Either way, the comment before the specific type annotation is more readable IMO so we'll go with that.

I just noticed this now: shouldn't 
```
    # file
    typing.IO[typing.AnyStr],
```
be
```
   # file
   typing.Union[typing.IO[typing.AnyStr], typing.AnyStr]
```
Since we do allow strings instead of IO objects?